core/trino-parser/src/main/java/io/trino/sql/tree/ArithmeticBinaryExpression.java

```diff
@@ -45,6 +45,7 @@ public String getValue()
     private final Operator operator;
     private final Expression left;
     private final Expression right;
+    private int hash;
```

Comment:
This needs documentation on why it’s safe to do this for an otherwise immutable class with respect to Java Memory model semantics.

Reply:
Is it enough to put this info in commit message?